cl/api/views.py

```diff
@@ -192,21 +191,19 @@ def coverage_data_opinions(request: HttpRequest):
         group_dict = dict(Court.JURISDICTIONS)
         if query_parameters:
             for court_name, court_id in query_parameters:
-                dates = fetch_start_end_dates_for_court(
-                    court_id=str(court_id),
-                )
-                if not dates:
+                first, last = fetch_start_end_dates_for_court(str(court_id))
+                if not first:
                     continue
-                group = dates.first()[1]
+                group = group_dict[Court.objects.get(id=court_id).jurisdiction]
                 court_data = {
                     "id": court_id,
                     "label": court_name,
                     "data": [
                         {
                             "val": court_id,
                             "timeRange": [
-                                dates.first()[0],
-                                dates.last()[0],
+                                first.date_filed,
+                                last.date_filed,
```

Comment:
But why not do the date part up above so the function fulfills its contract? It promises dates. It should return dates, right?

Reply:
Yep - this was refactored to include the dates in the function